src/bin/nydus-image/main.rs

```diff
@@ -126,16 +128,14 @@ impl OutputSerializer {
                 .write(true)
                 .open(f)
                 .with_context(|| format!("can not open output file {}", f.display()))?;
-            let trace = root_tracer!().dump_summary_map().unwrap_or_default();
-            let version = format!("{}-{}", build_info.package_ver, build_info.git_commit);
-            let output = Self {
-                version,
-                bootstrap: bootstrap.display().to_string(),
-                blobs: blob_ids,
-                trace,
-            };
-
-            serde_json::to_writer(w, &output).context("failed to write result to output file")?;
+            let mut out = serde_json::Map::new();
+            out.insert("version".to_string(), serde_json::Value::from(format!("{}-{}", build_info.package_ver, build_info.git_commit)));
+            out.insert("bootstrap".to_string(), serde_json::Value::from(bootstrap.display().to_string()));
+            out.insert("blobs".to_string(), serde_json::Value::from(blob_ids));
+            out.insert("trace".to_string(), serde_json::Value::from(root_tracer!().dump_summary_map().unwrap_or_default()));
+            out.insert("compressor".to_string(), serde_json::Value::from(compressor));
```

Comment:
Let's we define the fields in `OutputSerializer`.

Reply:
how about `fs_version` and `compressor`

Reply:
> how about `fs_version` and `compressor`

Should be ok.